modules/compiler/spirv-ll/test/spvasm/debug_info_funcs_and_blocks.spvasm

```diff
@@ -0,0 +1,76 @@
+; Copyright (C) Codeplay Software Limited
+;
+; Licensed under the Apache License, Version 2.0 (the "License") with LLVM
+; Exceptions; you may not use this file except in compliance with the License.
+; You may obtain a copy of the License at
+;
+;     https://github.com/codeplaysoftware/oneapi-construction-kit/blob/main/LICENSE.txt
+;
+; Unless required by applicable law or agreed to in writing, software
+; distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+; WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+; License for the specific language governing permissions and limitations
+; under the License.
+;
+; SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+; RUN: %if online-spirv-as %{ spirv-as --target-env %spv_tgt_env -o %spv_file_s %s %}
+; RUN: %if online-spirv-as %{ spirv-val %spv_file_s %}
```

Comment:
if we don't have online-spirv-as, then presumably this needs the offline spirv flag and all would be well?

Reply:
I'm not sure what you mean by "the offline spirv flag", sorry.

If the `online-spirv-as` isn't set then the test will use any pre-existing `%spv_file_s` file living in the build directory, which should normally have been configured by CMake to rebuild when you do `ninja spirv-ll-spvasm-lit` or `ninja check-spirv-ll-lit` or another target that depends on the files. Or not if you've misconfigured something or are running LIT yourself in a special way, but in that case the test will fail. This is how all of our `spirv-ll` LIT tests work, currently.